src/extension.ts

```diff
@@ -92,15 +92,17 @@ class FileBrowser {
     async update() {
         this.current.enabled = false;
         this.current.title = this.path.fsPath;
-        this.current.value = "";
+        // this.current.value = "";
 
         const stat = (await Result.try(vscode.workspace.fs.stat(this.path.uri))).unwrap();
         if (stat && this.inActions && (stat.type & FileType.File) === FileType.File) {
+            const selectedFile = new Path(this.path.uri).pop().unwrap();
             this.items = [
-                action("$(file) Open this file", Action.OpenFile),
-                action("$(split-horizontal) Open this file to the side", Action.OpenFileBeside),
-                action("$(edit) Rename this file", Action.RenameFile),
-                action("$(trash) Delete this file", Action.DeleteFile),
+                action(`$(new-file) Create file: ${this.current.value}`, Action.NewFile, this.current.value),
```

Comment:
Found an issue with this code. but am not able to update another CR to this branch. Hence mentioning it here:

Modification:
line#95: Should remain as is. However, I would create a copy  of the variable before resetting its value.

line#101: Use the new variable created on line#95